source/sigv4.c

```diff
@@ -2812,15 +2867,25 @@ static SigV4Status_t generateSigningKey( const SigV4Parameters_t * pSigV4Params,
                                    pSigV4Params->pCryptoInterface );
     }
 
-    if( hmacStatus == 0 )
+    if( isBufferSpaceSufficient && ( hmacStatus == 0 ) )
     {
         pSigningKey->pData = pSigningKeyStart;
         pSigningKey->dataLen = pSigV4Params->pCryptoInterface->hashDigestLen;
     }
-    else
+
+    /* Set the appropriate error code for failures. */
+    if( !isBufferSpaceSufficient )
+    {
+        returnStatus = SigV4InsufficientMemory;
+    }
+    else if( hmacStatus != 0 )
     {
         returnStatus = SigV4HashError;
     }
+    else
+    {
+        /* Empty else for MISRA C:2012 compliance. */
+    }
```

Comment:
Since `isBufferSpaceSufficient` is only ever set once, you could just set the return status at the same time. Then this block could just be
```suggestion
    /* Set the error code for hash failures. */
    if( hmacStatus != 0 )
    {
        returnStatus = SigV4HashError;
    }
```
and you wouldn’t need the empty else